frontend/src/features/public-form/components/FormPaymentPage/components/ProductPaymentItemDetailsBlock.tsx

```diff
@@ -107,15 +107,38 @@ const PaymentItem = ({
   )
 }
 
+const isProductSelected = (productItems: Array<ProductItem>) => {
+  let isProductSelected = false
+  for (let i = 0; i < productItems.length; i++) {
+    if (productItems[i].selected) {
+      isProductSelected = true
+      break
+    }
+  }
+  return isProductSelected
```

Comment:
Might be able to be terser with
```
const isSelected = productItems.some(item => item.selected)
```

Reply:
Thanks! [8c1861b](https://github.com/opengovsg/FormSG/pull/6585/commits/8c1861b8a943ef59e136a4211c708cac38859813)